### PR TITLE
Fix for issue #1014

### PR DESCRIPTION
**Pull request by mcarlson801**
This fixes the improper initialization of an array of ScalarT in Enthalpy_BasalResid and should fix the error found in issue #1014.

### CODE REVIEW

File: src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp [hunk 1]
```diff
@@ -81,9 +81,9 @@ operator() (const Enthalpy_Basal_Residual_Tag& tag, const int& sideSet_idx) cons
   const int cell = sideSet.ws_elem_idx.d_view(sideSet_idx);
   const int side = sideSet.side_pos.d_view(sideSet_idx);
 
-  ScalarT val[maxNumNodesPerSide];
+  ScalarT val[maxNumNodesPerSide] = {};
   for (unsigned int node = 0; node < numSideNodes; ++node) {
-      val[node] = 0;
+      val[node] = ScalarT(0);
```
Comment (mperego) [inline src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp:86]: I would remove this line.

Comment (mcarlson801) [reply src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp]: Good call, it's removed now

**Review by mperego — APPROVED** [review]
Looks good to me. Thanks!

File: src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp [hunk 2]
```diff
@@ -81,9 +81,8 @@ operator() (const Enthalpy_Basal_Residual_Tag& tag, const int& sideSet_idx) cons
   const int cell = sideSet.ws_elem_idx.d_view(sideSet_idx);
   const int side = sideSet.side_pos.d_view(sideSet_idx);
 
-  ScalarT val[maxNumNodesPerSide];
+  ScalarT val[maxNumNodesPerSide] = {};
   for (unsigned int node = 0; node < numSideNodes; ++node) {
-      val[node] = 0;
```
Comment (bartgol) [inline src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp:86]: Are we 100% sure that the default ctor of a sacado type is to init it as if with ScalarT(0)? If so, then this is fine.

**Review by bartgol — APPROVED** [review]

**Review by jewatkins — APPROVED** [review]

File: src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp (hunk 2 above)
Comment (mcarlson801) [reply src/landIce/evaluators/LandIce_EnthalpyBasalResid_Def.hpp:86]: Looking at the docs for DFad and SimpleFad, it looks like the default constructors do indeed initialize to 0.

**Pull request merged**